### PR TITLE
fix more bugs

### PR DESCRIPTION
**Pull request by AvihayMReblaze**

### CODE REVIEW

File: src/components/SideMenu.vue [hunk 1]
```diff
@@ -285,10 +285,16 @@ export default defineComponent({
         {
           href: `/${this.selectedBranch?.id}/account`,
           title: 'Account',
+          disabled: !this.userStore.checkAccessLevel(
+            [this.userStore.accessLevels.reblazeOrganizationAdmin, this.userStore.accessLevels.reblazeAdmin],
+          ) && this.userStore.getCookieProfileProvider() === 'reblaze',
```
Comment (Aviv-Galmidi) [inline src/components/SideMenu.vue:290]:
```suggestion
          disabled: !this.userStore.checkAccessLevel(
            [this.userStore.accessLevels.reblazeOrganizationAdmin, this.userStore.accessLevels.reblazeAdmin],
          ) || this.userStore.getCookieProfileProvider() !== 'reblaze',
```

The page is disabled if the user doesn't have enough access
```
!this.userStore.checkAccessLevel(
            [this.userStore.accessLevels.reblazeOrganizationAdmin, this.userStore.accessLevels.reblazeAdmin],
          )
```
OR
isn't connected through our provider
```
this.userStore.getCookieProfileProvider() !== 'reblaze'
```

File: src/components/SideMenu.vue [hunk 2]
```diff
@@ -285,10 +285,16 @@ export default defineComponent({
...
         },
         {
           href: `/${this.selectedBranch?.id}/users-management`,
           title: 'Users Management',
+          disabled: !this.userStore.checkAccessLevel(
+            [this.userStore.accessLevels.reblazeOrganizationAdmin, this.userStore.accessLevels.reblazeAdmin],
+          ) && this.userStore.getCookieProfileProvider() === 'reblaze',
```
Comment (Aviv-Galmidi) [inline src/components/SideMenu.vue:297]:
```suggestion
          disabled: !this.userStore.checkAccessLevel(
            [this.userStore.accessLevels.reblazeOrganizationAdmin, this.userStore.accessLevels.reblazeAdmin],
          ) || this.userStore.getCookieProfileProvider() !== 'reblaze',
```

File: src/stores/userStore.ts [hunk 3]
```diff
@@ -246,6 +246,12 @@ export const useUserStore = defineStore('user', () => {
     return userCookieValue.email
   }
 
+  function getCookieProfileProvider(): string {
+    // Cannot be a computed value as cookies are not reactive
+    const userCookieValue = getUserCookieValue()
+    return userCookieValue.profile.provider
```
Comment (Aviv-Galmidi) [inline src/stores/userStore.ts:252]:
```suggestion
    return userCookieValue.profile?.provider
```

Let's protect ourselves from reading an attribute of undefined

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

File: src/components/SideMenu.vue (hunk 1 above)
Comment (AvihayMReblaze) [reply src/components/SideMenu.vue:290]: Right, done.

File: src/components/SideMenu.vue (hunk 2 above)
Comment (AvihayMReblaze) [reply src/components/SideMenu.vue:297]: Done

File: src/stores/userStore.ts (hunk 3 above)
Comment (AvihayMReblaze) [reply src/stores/userStore.ts]: Done

**Review by Aviv-Galmidi — APPROVED** [review]

**Pull request merged**